### PR TITLE
Addition of release process doc

### PR DESCRIPTION
**Pull request by dannon**
As mentioned in the doc, this just outlines the release cycle to set expectations.

This is nothing new in terms of policy, it's just trying to document the current state of things.

*Please* feel free to directly push any edits/additions/etc.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: doc/source/project/releases.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+This document outlines the Galaxy release cycle so administrators and users know
+when to expect updates and what each update includes, making it easier to plan
+for upgrades and maintain their systems.  It does not cover the actual execution
+of the release process, which is covered elsewhere. (link)
+
+
+Major or Long Term Support (LTS) Releases
+-----------------------------------------
+
+The first release of each year is Galaxy's annual LTS release. The LTS release
```
Comment (nsoranzo) [inline doc/source/project/releases.rst]: Maybe we can say since when we started this convention (23.0?).

File: doc/source/project/releases.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+is a major version update corresponding to the year, for example, 24.0.  This
+release receives significant and thorough testing by a dedicated team that does
+end-to-end testing of a variety of analysis protocols and will be supported and
```
Comment (jdavcs) [inline doc/source/project/releases.rst]:
```suggestion
release receives significant and thorough end-to-end testing by a dedicated team and will be supported and
```

Comment (jdavcs) [reply doc/source/project/releases.rst]: Or better, "thorough testing", i.e., no "end-to-end". I'd keep it open.

File: doc/source/project/releases.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+receive bugfixes until the next LTS.  As such, this release is ideally targeted
+for production use by local Galaxy administrators.
+
+
+Minor Releases
+--------------
+
+Throughout the year, we also ship several (currently 2-3) minor releases. These
```
Comment (jdavcs) [inline doc/source/project/releases.rst]:
```suggestion
Throughout the year, we also ship several minor releases. These
```
or it'll tie us to 2-3

File: doc/source/project/releases.rst (hunk 2 above)
Comment (dannon) [reply doc/source/project/releases.rst]: @jdavcs Please feel free to push whatever tweaks you want

File: doc/source/project/releases.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+Our release schedule is as follows:
+
+- LTS Release: Annually, in the first quarter.
+- Minor Releases: 2-3 times a year, generally in the early summer and fall.
+- Point Releases: As needed.
+
+Please note that the release dates can vary based on the development process and
+testing results.
+
+The usegalaxy.* public servers will be updated to the latest release within a
```
Comment (martenson) [inline doc/source/project/releases.rst]: We should double check with EU/AU admins before we make such a promise.

File: doc/source/project/releases.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+Minor Releases
+--------------
+
+Throughout the year, we also ship several (currently 2-3) minor releases. These
+are version updates like 24.1, 24.2.  These can also include significant
+updates, new features, etc., but do not follow the full release-testing protocol
+as with the LTS release.  These are targeted for administrators who want to stay
+up to date with the latest features and improvements.  Usegalaxy.org deploys new
+minor releases as a part of the release process, and other usegalaxy.* servers
+should strive to stay up to date with these.
```
Comment (martenson) [inline doc/source/project/releases.rst]: Ditto here, EU/AU should confirm this statement.

File: doc/source/project/releases.rst (hunk 4 above)
Comment (martenson) [reply doc/source/project/releases.rst]: ping @bgruening @cat-bro

Comment (dannon) [reply doc/source/project/releases.rst]: xref https://galaxyproject.org/usegalaxy/#membership, which says 90 days -- we should make sure these align.

edit '90 business days', I guess, which is more like 4-5 months.  In any event, we should make sure this is correct and in alignment.

File: doc/source/project/releases.rst (hunk 3 above)
Comment (dannon) [reply doc/source/project/releases.rst]: I don't think this ties us to 2-3 at all, just sets expectations by communicating the current frequency?  If this is confusing though, we can drop it of course.

**Comment by jdavcs** [comment]
To echo Martin's comment, and given our today's discussion (the one in the UI/UX channel thread)  I think it would make sense to postpone this until we can have a larger meeting, with EU/AU folks present, where we can discuss in more detail the release life cycle, formal testing of releases, and any related issues. That said, this document should absolutely be part of our documentation, I'm sure it'll be most helpful for the community.

File: doc/source/project/releases.rst (hunk 3 above)
Comment (martenson) [reply doc/source/project/releases.rst]: "usually" ?

**Review by jdavcs — APPROVED** [review]

**Comment by martenson** [comment]
It might be beneficial to clarify the security commitment wrt what has been added here and also the release versioning. e.g. the following seems confusing now:

https://github.com/galaxyproject/galaxy/blob/8adfb7e7b5cf00dcbc4d6ee46986dd4539bed853/SECURITY.md?plain=1#L13-L20

**Pull request merged**